cpp/src/slice2js/Gen.cpp

```diff
@@ -2268,234 +2215,231 @@ Slice::Gen::TypeScriptImportVisitor::visitDictionary(const DictionaryPtr& dict)
-    //
-    string::size_type i = type.find(".");
-    if (prefix.empty() && i != string::npos)
+    if (!type)
     {
-        if (scope.find("." + type.substr(0, i + 1)) != string::npos)
+        assert(!optional);
+        return "void";
+    }
+
+    string t;
+
+    static const char* typeScriptBuiltinTable[] = {
+        "number",   // byte
+        "boolean",  // bool
+        "number",   // short
+        "number",   // int
+        "Ice.Long", // long
+        "number",   // float
+        "number",   // double
+        "string",
+        "Ice.Value", // Ice.Object
+        "Ice.ObjectPrx",
+        "Ice.Value"};
+
+    BuiltinPtr builtin = dynamic_pointer_cast<Builtin>(type);
+    if (builtin)
+    {
+        switch (builtin->kind())
         {
-            for (vector<pair<string, string>>::const_iterator j = _aliases.begin(); j != _aliases.end(); ++j)
+            case Builtin::KindLong:
             {
-                if (j->first == type)
+                t = _iceImportPrefix + typeScriptBuiltinTable[builtin->kind()];
+                // For Slice long parameters we accept both number and Ice.Long, this is more convenient
+                // for the user as he can use the native number type when the value is within the range of
+                // number (2^53 - 1) and Ice.Long when the value is outside of this range.
+                if (forParameter)
                 {
-                    return;
+                    t += " | number";
                 }
+                break;
+            }
+            case Builtin::KindObject:
+            case Builtin::KindObjectProxy:
+            case Builtin::KindValue:
+            {
+                t = _iceImportPrefix + typeScriptBuiltinTable[builtin->kind()];
+                if (nullable)
+                {
+                    t = _iceImportPrefix + "Ice.Nullable<" + t + ">";
+                }
+                break;
+            }
+            default:
+            {
+                t = typeScriptBuiltinTable[builtin->kind()];
+                break;
             }
-            string alias = type;
-            replace(alias.begin(), alias.end(), '.', '_');
-            //
-            // We prefix alias with iceA this avoid conflict with iceNSX used for
-            // import prefixes
-            //
-            _aliases.push_back(make_pair(type, "iceA_" + alias));
         }
     }
-}
 
-bool
-Slice::Gen::TypeScriptAliasVisitor::visitClassDefStart(const ClassDefPtr& p)
-{
-    ModulePtr module = dynamic_pointer_cast<Module>(p->container());
-    //
-    // Add alias required for base class
-    //
-    ClassDefPtr base = p->base();
-    if (base)
+    ClassDeclPtr cl = dynamic_pointer_cast<ClassDecl>(type);
+    if (cl)
     {
-        addAlias(dynamic_pointer_cast<Type>(base->declaration()), module);
+        t = importPrefix(cl->scoped()) + fixId(cl->scoped());
+        if (nullable)
+        {
+            t = _iceImportPrefix + "Ice.Nullable<" + t + ">";
```

Comment:
We have our own nullable? We should probably get rid of this we can,

Reply:
Is an alias for `T | null`

Reply:
Why have it though?

Reply:
Not strong feelings here, we can go either way. Is just a minor detail.

Reply:
I will fix it in a follow-up PR.